howdoi/howdoi.py

```diff
@@ -66,22 +65,40 @@ def u(x):
     'bing': SCHEME + 'www.bing.com/search?q=site:{0}%20{1}',
     'google': SCHEME + 'www.google.com/search?q=site:{0}%20{1}'
 }
+
+BLOCK_INDICATORS = (
+        'form id="captcha-form"',
+        'This page appears when Google automatically detects requests coming from your computer network which appear to be in violation of the <a href="//www.google.com/policies/terms/">Terms of Service'
+        )
+
 STAR_HEADER = u('\u2605')
 ANSWER_HEADER = u('{2}  Answer from {0} {2}\n{1}')
 NO_ANSWER_MSG = '< no answer given >'
-XDG_CACHE_DIR = os.environ.get('XDG_CACHE_HOME',
-                               os.path.join(os.path.expanduser('~'), '.cache'))
-CACHE_DIR = os.path.join(XDG_CACHE_DIR, 'howdoi')
-CACHE_FILE = os.path.join(CACHE_DIR, 'cache{0}'.format(
-    sys.version_info[0] if sys.version_info[0] == 3 else ''))
+
+CACHE_DIR = appdirs.user_cache_dir('howdoi')
+CACHE_ENTRY_MAX = 128
 
 if os.getenv('HOWDOI_DISABLE_CACHE'):
-    howdoi_session = requests.session()
+    cache = NullCache() # works like an always empty cache, cleaner than 'if cache:' everywhere
 else:
-    if not os.path.exists(CACHE_DIR):
-        os.makedirs(CACHE_DIR)
-    howdoi_session = requests_cache.CachedSession(CACHE_FILE)
+    cache = FileSystemCache(CACHE_DIR, CACHE_ENTRY_MAX, default_timeout=0)
```

Comment:
Do we want to allow for cache expiration after some amount of time? Perhaps it's 0 by default, but eventually expires since StackOverflow answers are constantly changing and getting better.

Reply:
Keep in mind that it's a rolling cache with 128 entries. Each new invocation creates at least 3 + num_answers. I don't expect the cache to get stale. With that said, it's easy to set; we can even set different TTLs on output/links and answer-html entries (there is a timeout option on cache.set()). If you come up with values, I'll put them in.

Reply:
Didn't realize it was rolling. This works.

Reply:
Yes, it's LIFO pruned to CACHE_ENTRY_MAX. The pruning is actually what is O(n) - and what I would fix first (they pickle modification date, so have to open file to check; it should be done with stat() instead, with dir still in cache it wouldn't hit disk).